cmd/mailbox-controller/controller.go

```diff
@@ -255,76 +258,82 @@ func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 func (ctl *mbCtl) ensureBinding(ctx context.Context, workspace *tenancyv1alpha1.Workspace) bool {
 	logger := klog.FromContext(ctx).WithValues("mbsName", workspace.Name)
 
-	// The script ensures that a mailbox workspace has the needed KubeStellar CRDs.
 	// The script must be idempotent.
-	shellScriptName := "kubectl-kubestellar-ensure-mbw"
+	shellScriptName := "kube-bind"
 
-	logger.V(2).Info("Executing shell script", "script", shellScriptName)
-	// Make a sandbox for the concurrent access of the kubeconfig from the mailbox-controller
-	cmdLine := "cp $KUBECONFIG $KUBECONFIG.copy && KUBECONFIG=$KUBECONFIG.copy " + shellScriptName + " " + workspace.Name
-	logger.V(2).Info(cmdLine)
-	cmd := exec.Command("/bin/sh", "-c", cmdLine)
-	stdout, err := cmd.StdoutPipe()
-	if err != nil {
-		logger.Error(err, "Unable to return a pipe for stdout")
-		return true
-	}
-	stderr, err := cmd.StderrPipe()
-	if err != nil {
-		logger.Error(err, "Unable to return a pipe for stderr")
-		return true
-	}
-	err = cmd.Start()
-	if err != nil {
-		logger.Error(err, "Unable to start executing the script")
-		return true
-	}
-	outBuf := bufio.NewReader(stdout)
-	errBuf := bufio.NewReader(stderr)
+	resourcesToBind := []string{"syncerconfigs", "edgesyncconfigs"}
+	for idx, resource := range resourcesToBind {
+		logger.V(2).Info("Ensuring binding", "script", shellScriptName, "resource", resource)
 
-	r, _ := regexp.Compile("🔒 (Created|Updated) secret.*namespace (.*)")
-	found, namespaceName := false, ""
-	for {
-		line, _, err := errBuf.ReadLine()
+		// suffix helps isolate this controller's multiple workers to prevent concurrent access of a single kubeconfig file.
+		// This is a compromise due to the situation that 1) we are using kcp and 2) we are running tests on bare process of this controller.
```

Comment:
Part 2 is not important here. Even when run in a container, the mailbox controller may be doing concurrent script invocations using the same filesystem.

Reply:
Changed in latest push.